warehouse/query-core/src/main/java/datawave/query/function/serdes/AttributeKryo.java

```diff
@@ -0,0 +1,88 @@
+package datawave.query.function.serdes;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import datawave.query.attributes.Attribute;
+import datawave.query.attributes.DocumentPayload;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Consumer;
+
+public class AttributeKryo {
+
+    protected Kryo kryo;
+    boolean reducedSize;
+
+    protected AttributeKryo(Kryo kryo) {
+        this.kryo = kryo;
+    }
+    public static class Serializer extends AttributeKryo implements Consumer<Map.Entry<String,Attribute<? extends Comparable<?>>>> {
+
+        protected Output output;
+        public Serializer(Kryo kryo, Output output) {
+            super(kryo);
+            this.output = output;
+        }
+
+        public void accept(Map.Entry<String,Attribute<? extends Comparable<?>>> entry) {
+
+            output.writeString(entry.getKey());
+
+            Attribute<?> attribute = entry.getValue();
+            output.writeString(attribute.getClass().getName());
+            attribute.write(kryo, output, reducedSize);
+        }
+    }
+
+    public static class Deserializer extends AttributeKryo implements Consumer<Input> {
+
+        DocumentPayload documentPayload;
+        TreeMap<String, Attribute<?>> dictionary;
+
+        public Deserializer(Kryo kryo, DocumentPayload documentPayload) {
+            super(kryo);
+            this.documentPayload = documentPayload;
+            this.dictionary = documentPayload.getDictionary();
+        }
+
+        public void accept(Input input) {
+            String fieldName = input.readString();
+
+            // Get the class name for the concrete Attribute
+            String attrClassName = input.readString();
```

Comment:
for the future I ran a few tests and using a caffeine cache loader for the class and constructors reduced the time by a 1/4